python/metatomic_torch/metatomic/torch/ase_calculator.py

```diff
@@ -869,3 +868,521 @@ def _get_energy_uncertainty_output():
+    Lebedev (S^2) x Uniform (S^1) grid of rotations in SO(3).
+    """
+
+    implemented_properties = ["energy", "forces", "stress"]
+
+    def __init__(
+        self,
+        base_calculator: MetatomicCalculator,
+        lebedev_order: int = 3,
+        n_inplane_rotations: int = 4,
+        batch_size: Optional[int] = None,
+        **kwargs,
+    ):
+        try:
+            from scipy.integrate import lebedev_rule  # noqa: F401
+        except ImportError as e:
+            raise ImportError(
+                "scipy is required to use the SO3AveragedCalculator, please install "
+                "it with `pip install scipy` or `conda install scipy`"
+            ) from e
+
+        super().__init__(**kwargs)
+
+        self.base_calculator = base_calculator
+        self.lebedev_order = lebedev_order
+        self.n_inplane_rotations = n_inplane_rotations
+
+        self.so3_quadrature_rotations = _get_so3_quadrature(
+            lebedev_order, n_inplane_rotations
+        )
+
+        self.batch_size = (
+            batch_size if batch_size is not None else len(self.so3_quadrature_rotations)
+        )
+
+    def calculate(self, atoms, properties, system_changes):
+        super().calculate(atoms, properties, system_changes)
+
+        compute_forces_and_stresses = "forces" in properties or "stress" in properties
+
+        if len(self.so3_quadrature_rotations) > 0:
+            rotated_atoms_list = _rotate_atoms(atoms, self.so3_quadrature_rotations)
+            batch_size = (
+                self.batch_size
+                if self.batch_size is not None
+                else len(rotated_atoms_list)
+            )
+            batches = [
+                rotated_atoms_list[i : i + batch_size]
+                for i in range(0, len(rotated_atoms_list), batch_size)
+            ]
+            results: Dict[str, np.ndarray] = {}
+            for batch in batches:
+                try:
+                    batch_results = self.base_calculator.compute_energy(
+                        batch, compute_forces_and_stresses
+                    )
+                    for key, value in batch_results.items():
+                        results.setdefault(key, [])
+                        results[key].extend(
+                            [value] if isinstance(value, float) else value
+                        )
+                except torch.cuda.OutOfMemoryError as e:
+                    raise RuntimeError(
+                        "Out of memory error encountered during rotational averaging. "
+                        "Please reduce the batch size or use lower rotational "
+                        "averaging parameters. This can be done by setting the "
+                        "`batch_size`, `lebedev_order`, and `n_inplane_rotations` "
+                        "parameters while initializing the calculator."
+                        f"Full error message: {e}"
+                    )
```

Comment:
```suggestion
                    raise RuntimeError(
                        "Out of memory error encountered during rotational averaging. "
                        "Please reduce the batch size or use lower rotational "
                        "averaging parameters. This can be done by setting the "
                        "`batch_size`, `lebedev_order`, and `n_inplane_rotations` "
                        "parameters while initializing the calculator."
                    ) from e
```

`raise Exception(...) from e` is a nicer way to keep the old exception around!